### DISCUSSION TITLE
How to include the installation parameters and make BLAS arguments in the requirements.txt?

### DISCUSSION THREAD

**Discussion by AayushSameerShah** [abetlen/llama-cpp-python#976] [Q&A] [answered]
# 🙋🏻‍♂️ 
I am building a simple project in my local machine **and then will deploy on AWS ubuntu server as an API**.
👉🏻 In the development environment I have installed this application using:
```sh
$env:CMAKE_ARGS = "-DLLAMA_BLAS=ON -DLLAMA_BLAS_VENDOR=OpenBLAS"
pip install llama-cpp-python
```
👉🏻 When I do the `pip freeze > requirements.txt` *(obviously)* it is not going to include these env arguments.

### ⁉ Question
*How can In include these arguments in the requirements so that in the deployment environment I don't have to configure things manually?* - **And is that even possible?**

**Comment by abetlen** [comment, marked as answer]
Cross-posted my answer from the issue.

@AayushSameerShah late reply but I have a bettter answer for you! In newer versions of `pip` it's possible to specify `-C/--config-settings` flag which passes args to the build tool (scikit-build-core in our case) so you can do

```bash
# requirements.txt
llama-cpp-python -C cmake.args="-DLLAMA_BLAS=ON;-DLLAMA_BLAS_VENDOR=OpenBLAS"
```

This is also documented now in the README.md now under `CLI / requirements.txt`. You may need to run `pip install --upgrade pip` if you get an error about the `-C` option to upgrade to a newer version of pip.

Hope this helps.
